frontend/src/metabase/query_builder/components/notebook/steps/AggregateStep/AggregateStep.tsx

```diff
@@ -21,21 +21,24 @@ export function AggregateStep({
     return Lib.aggregations(query, stageIndex);
   }, [query, stageIndex]);
 
-  const handleAddAggregation = (aggregation: Lib.Aggregable) => {
-    const nextQuery = Lib.aggregate(query, stageIndex, aggregation);
+  const handleAddAggregations = (aggregations: Lib.Aggregable[]) => {
+    const nextQuery = aggregations.reduce(
+      (query, aggregation) => Lib.aggregate(query, stageIndex, aggregation),
+      query,
+    );
     updateQuery(nextQuery);
   };
 
-  const handleUpdateAggregation = (
+  const handleUpdateAggregations = (
     currentClause: Lib.AggregationClause,
-    nextClause: Lib.Aggregable,
+    nextClauses: Lib.Aggregable[],
   ) => {
-    const nextQuery = Lib.replaceClause(
-      query,
-      stageIndex,
-      currentClause,
-      nextClause,
+    const [firstClause, ...remainingClauses] = nextClauses;
+    const nextQuery = remainingClauses.reduce(
+      (query, aggregation) => Lib.aggregate(query, stageIndex, aggregation),
```

Comment:
Something is wrong here. It's a combination of add + update.

Reply:
It's confusing but somewhat intended - if you're editing an aggregation you are able to turn it into up to 3 aggregations by going through the new shortcut that this PR adds. So we need to replace + add.

Maybe we should not show the shortcut when editing, I'll double-check with the product team.

Reply:
There's been a decision that the new "Compare to previous rows" option should always produce new aggregations and not affect existing ones: 07e0153ab72e445325647c13b4eb720a893ab942  + daec4e5658da3901a518c73cde1136d40ca569c2